mongo/integration/errors_test.go

```diff
@@ -289,4 +307,121 @@ func TestErrors(t *testing.T) {
 			})
 		}
 	})
+	mt.Run("error helpers", func(mt *mtest.T) {
+		//IsDuplicateKeyError
+		mt.Run("IsDuplicateKeyError", func(mt *mtest.T) {
+			testCases := []struct {
+				name   string
+				err    error
+				result bool
+			}{
+				{"CommandError true", mongo.CommandError{11000, "", nil, "blah", nil}, true},
+				{"CommandError false", mongo.CommandError{100, "", nil, "blah", nil}, false},
+				{
+					"WriteException true in writeConcernError",
+					mongo.WriteException{
+						&mongo.WriteConcernError{"name", 11001, "bar", nil},
+						mongo.WriteErrors{
+							mongo.WriteError{0, 100, "baz"},
+						},
+						nil,
+					},
+					true,
+				},
+				{
+					"WriteException true in writeErrors",
+					mongo.WriteException{
+						&mongo.WriteConcernError{"name", 100, "bar", nil},
+						mongo.WriteErrors{
+							mongo.WriteError{0, 12582, "baz"},
+						},
+						nil,
+					},
+					true,
+				},
+				{
+					"WriteException false",
+					mongo.WriteException{
+						&mongo.WriteConcernError{"name", 16460, "bar", nil},
+						mongo.WriteErrors{
+							mongo.WriteError{0, 100, "blah  E11000 blah"},
+						},
+						nil,
+					},
+					false,
+				},
+				{
+					"BulkWriteException true",
+					mongo.BulkWriteException{
+						&mongo.WriteConcernError{"name", 100, "bar", nil},
+						[]mongo.BulkWriteError{
+							{mongo.WriteError{0, 16460, "blah  E11000 blah"}, &mongo.InsertOneModel{}},
+						},
+						[]string{"otherError"},
+					},
+					true,
+				},
+				{
+					"BulkWriteException false",
+					mongo.BulkWriteException{
+						&mongo.WriteConcernError{"name", 100, "bar", nil},
+						[]mongo.BulkWriteError{
+							{mongo.WriteError{0, 110, "blah"}, &mongo.InsertOneModel{}},
+						},
+						[]string{"otherError"},
+					},
+					true,
```

Comment:
You probably saw this, but I think this should be `false`.

Reply:
yup, thanks!